### DISCUSSION TITLE
The PWM on pin D2 does not work (LGT8F328P)

### DISCUSSION THREAD

**Discussion by olfresh-cf** [dbuezas/lgt8fx#113] [General]
Hi,
The PWM on pin D2 does not work. Switches only to 0 or 1 when using analogWrite(D2, value). If you use the original kernel https://github.com/LGTMCU/Larduino_HSP, then the PWM on pin D2 works with the pwmWrite(D2, value) command. It looks like the difference is in a piece of code in the kernel file wiring_digital.c:
[https://github.com/LGTMCU/Larduino_HSP]:
```
void pwmWrite(uint8_t pin, uint16_t val) {... 
  #if defined(TCCR3A) && defined(COM3B1)
  case TIMER3BX: // F2 tied with D2
  #if defined(__LGT8FX8P32__)
  case TIMER3B:
  cbi(DDRD, PD2);
  #endif
  // connect pwm to pin on timer 3, channel B
  //OCR3B = val; // set pwm duty
  atomicWriteWord(&OCR3BL, val);
  sbi(TCCR3A, COM3B1);
  sbi(DDRF, PF2);
  break;
#endif ....}
``` 
[https://github.com/dbuezas/lgt8fx/]:
```
void analogWrite(uint8_t pin, int val){... 
  #if defined(TCCR3A) && defined(COM3B1)
  case TIMER3B:
  // connect pwm to pin on timer 3, channel B
  sbi(TCCR3A, COM3B1);
  OCR3B = val; // set pwm duty
  break;
  #endif ...}
```

**Comment by LaZsolt** [comment]
@olfresh-cf:
> The PWM on pin D2 does not work.

I never try to use PWM, but I have an idea. PF2 arduino pin name is D34 so try this:
`analogWrite(D34, value);`
